File: SDKs/amazonaws.com/appmesh/2019-01-25/python/src/sdk/models/shared/virtualgatewaylogging.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import virtualgatewayaccesslog as shared_virtualgatewayaccesslog


@dataclass_json
@dataclasses.dataclass
class VirtualGatewayLogging:
    r"""VirtualGatewayLogging
    An object that represents logging information.
    """
    
    access_log: Optional[shared_virtualgatewayaccesslog.VirtualGatewayAccessLog] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('accessLog') }})
    

File: SDKs/amazonaws.com/appmesh/2019-01-25/java/lib/src/main/java/openapisdk/models/shared/VirtualGatewayLogging.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * VirtualGatewayLogging
 * An object that represents logging information.
**/
public class VirtualGatewayLogging {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("accessLog")
    public VirtualGatewayAccessLog accessLog;
    public VirtualGatewayLogging withAccessLog(VirtualGatewayAccessLog accessLog) {
        this.accessLog = accessLog;
        return this;
    }
}